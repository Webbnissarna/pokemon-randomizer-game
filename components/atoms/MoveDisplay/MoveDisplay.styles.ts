import type { ThemeUIStyleObject } from "theme-ui";

export const containerStyle: ThemeUIStyleObject = {
  flexDirection: "column",
  padding: "sm",
  fontSize: "sm",
};

export const innerStyle: ThemeUIStyleObject = { gap: "sm" };

export const nameTextStyle: ThemeUIStyleObject = { flexGrow: 1 };

export const textStyle: ThemeUIStyleObject = {
  textAlign: "left",
};
