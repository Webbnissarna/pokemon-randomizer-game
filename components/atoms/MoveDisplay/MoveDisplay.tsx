import { Flex, Box, Text } from "theme-ui";
import {
  TypeColors,
  getLatestUsefulMoveDescription,
} from "../../../utils/pokemonMeta";
import {
  containerStyle,
  innerStyle,
  nameTextStyle,
  textStyle,
} from "./MoveDisplay.styles";
import type { MoveDisplayProps } from "./MoveDisplay.types";

export default function MoveDisplay({ move }: MoveDisplayProps): JSX.Element {
  return (
    <Flex
      sx={{
        ...containerStyle,
        backgroundColor: TypeColors[move.type],
      }}
    >
      <Flex sx={innerStyle}>
        <Box>
          <Text>{move.type}</Text>
        </Box>
        <Text>{move.category}</Text>
        <Text sx={nameTextStyle}>{move.name.en}</Text>
        <Text>{move.power}</Text>
        <Text>{move.pp} PP</Text>
        <Text>{move.accuracy}%</Text>
      </Flex>
      <Text sx={textStyle}>
        {getLatestUsefulMoveDescription(move.description).desc}
      </Text>
    </Flex>
  );
}
