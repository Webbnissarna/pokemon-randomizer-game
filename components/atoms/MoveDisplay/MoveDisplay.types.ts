import type { MoveInfo } from "../../../utils/Generator/pokemon.types";

export interface MoveDisplayProps {
  move: MoveInfo;
}
