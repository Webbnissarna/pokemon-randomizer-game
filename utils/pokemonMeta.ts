import * as CSS from "csstype";

export enum Type {
  Bug = "Bug",
  Dark = "Dark",
  Dragon = "Dragon",
  Electric = "Electric",
  Fairy = "Fairy",
  Fighting = "Fighting",
  Fire = "Fire",
  Flying = "Flying",
  Ghost = "Ghost",
  Grass = "Grass",
  Ground = "Ground",
  Ice = "Ice",
  Normal = "Normal",
  Poison = "Poison",
  Psychic = "Psychic",
  Rock = "Rock",
  Steel = "Steel",
  Water = "Water",
}

export const TypeColor: Record<Type, CSS.Property.Color> = {
  Bug: "#A8B820",
  Dark: "#6A5D55",
  Dragon: "#7038F8",
  Electric: "#F8D030",
  Fairy: "#EE99AC",
  Fighting: "#C03028",
  Fire: "#F08030",
  Flying: "#A890F0",
  Ghost: "#705898",
  Grass: "#78C850",
  Ground: "#E0C068",
  Ice: "#98D8D8",
  Normal: "#A8A878",
  Poison: "#A040A0",
  Psychic: "#F85888",
  Rock: "#B8A038",
  Steel: "#B8B8D0",
  Water: "#6890F0",
};
