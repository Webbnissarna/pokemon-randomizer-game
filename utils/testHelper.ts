import type {
  GameProp,
  GamePropGetter,
} from "./GamePropSystem/GamePropSystem.types";
import type { GeneratedPokemon } from "./Generator/generator.types";
import type { PokemonInfo, Nature, MoveInfo } from "./Generator/pokemon.types";

import Pikachu from "../public/data/pokemon/25.json";
import Pound from "../public/data/moves/1.json";
import KarateChop from "../public/data/moves/2.json";
import Fly from "../public/data/moves/19.json";
import Thunder from "../public/data/moves/87.json";
import Natures from "../public/data/naturesList.json";
import Abilities from "../public/data/abilitiesList.json";

export const MOCK_PROP: Required<GameProp> = {
  mainProp: "mainMock",
  optionalProp: "optionalMock",
};

export const MOCK_NATURE = Natures[15] as Nature;

export const mockPropGetter: GamePropGetter<GeneratedPokemon> = (_) =>
  MOCK_PROP;

export const mockPokemon: GeneratedPokemon = {
  info: Pikachu as PokemonInfo,
  nature: MOCK_NATURE,
  showdownExport: "",
  ability: Abilities[0],
  moves: [Pound, KarateChop, Fly, Thunder] as MoveInfo[],
};
