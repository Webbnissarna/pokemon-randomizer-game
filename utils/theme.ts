import { Theme, useThemeUI, ThemeUIContextValue } from "theme-ui";

const makeTheme = <T extends Theme>(t: T) => t;

const theme = makeTheme({
  colors: {
    text: "#000",
    background:
      "linear-gradient(0deg, rgba(255,172,252,1) 0%, rgba(157,214,255,1) 100%);",

    ball: {
      active: {
        top: "#ff6060",
        middle: "#606060",
        bottom: "#ffffff",
      },
      disabled: {
        top: "#4f4f4f",
        middle: "#414141",
        bottom: "#838383",
      },
    },

    positive: "#73FF67",
    positiveText: "#333333",
    negative: "#B30000",
    negativeText: "#FFFFFF",
    neutral: "#67C8FF",
    neutralText: "#000000",
    secondary: "#4E7999",
    secondaryText: "#FFFFFF",
    tertiary: "#FFFFFF",
    tertiaryText: "#000000",

    inputBackground: "#D0D0D0",
    inputText: "#3D3D3D",

    contentBackground: "#FFFFFF",
  },
  fonts: {
    body: "'Noto Sans JP', sans-serif",
    heading: "'Noto Sans JP', sans-serif",
  },
  shadows: {
    sm: "0px 2px 2px 0px rgba(0, 0, 0, 0.25)",
    md: "0px 4px 4px 0px rgba(0, 0, 0, 0.25)",

    svgFilter: {
      sm: "drop-shadow(rgba(0, 0, 0, 0.25) 0px 2px 2px)",
      md: "drop-shadow(rgba(0, 0, 0, 0.25) 0px 4px 4px)",
    },
  },
  radii: {
    sm: 5,
    md: 10,
  },
  space: {
    md: 10,
    lg: 18,
    xl: 40,
  },
  sizes: {
    sm: 30,
    md: 40,
    lg: 60,
  },
  fontSizes: {
    xxs: 9,
    xs: 10,
    sm: 12,
    md: 14,
    lg: 18,
    xl: 24,
  },

  styles: {
    root: {
      fontFamily: "body",

      h1: {
        fontSize: "xl",
      },
    },
  },
});

export default theme;
export type ThemeType = typeof theme;

interface ExactContextValue extends Omit<ThemeUIContextValue, "theme"> {
  theme: ThemeType;
}

export const useTheme = useThemeUI as unknown as () => ExactContextValue;
