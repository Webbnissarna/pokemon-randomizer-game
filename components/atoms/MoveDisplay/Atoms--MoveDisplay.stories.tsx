import { Flex } from "theme-ui";
import MoveDisplay from ".";
import { MOCK_MOVES } from "../../../utils/testHelper";

export function Default() {
  return (
    <Flex sx={{ flexDirection: "column", gap: "lg" }}>
      {MOCK_MOVES.map((move) => (
        <MoveDisplay key={move.indexNo} move={move} />
      ))}
    </Flex>
  );
}
