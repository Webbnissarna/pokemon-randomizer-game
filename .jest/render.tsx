import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
