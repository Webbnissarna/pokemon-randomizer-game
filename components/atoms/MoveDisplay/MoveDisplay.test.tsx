import { render } from "../../../.jest/render";
import MoveDisplay from ".";
import { MOCK_MOVES } from "../../../utils/testHelper";

describe("MoveDisplay", () => {
  it("renders", () => {
    const { asFragment } = render(<MoveDisplay move={MOCK_MOVES[0]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
