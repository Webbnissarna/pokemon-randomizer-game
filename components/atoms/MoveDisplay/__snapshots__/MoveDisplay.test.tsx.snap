// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveDisplay renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-1usc76v-Box"
  >
    <div
      class="css-1siex4w-Box"
    >
      <div
        class="css-1peva9n-Box"
      >
        <span
          class="css-1peva9n-Box"
        >
          Normal
        </span>
      </div>
      <span
        class="css-1peva9n-Box"
      >
        Physical
      </span>
      <span
        class="css-8jjk1d-Box"
      >
        Pound
      </span>
      <span
        class="css-1peva9n-Box"
      >
        40
      </span>
      <span
        class="css-1peva9n-Box"
      >
        35 PP
      </span>
      <span
        class="css-1peva9n-Box"
      >
        100%
      </span>
    </div>
    <span
      class="css-gety84-Box"
    >
      The target is physically pounded with a long tail, a foreleg, or the like.
    </span>
  </div>
</DocumentFragment>
`;
